### PR TITLE
added digitalocean deployment instructions (fix #582)

### LINKED ISSUE #582: add deploying guide on digital ocean
Docs already lists deployment on [Kubernetes](https://github.com/hasura/graphql-engine/tree/master/docs/graphql/manual/deployment/kubernetes). Similarly, we need deployment guide on a Digital Ocean Droplet, using Docker Compose.

The guide should include instructions to update to latest version, as well as setting an access key. 

HTTPS setup can be found at https://github.com/hasura/graphql-engine/tree/master/install-manifests/docker-compose-https

### PR DESCRIPTION
**Pull request by iUtsavB**
### Description

Added step by step documentation for deployment on DigitalOcean using Docker 
The GraphQL Engine could be accessible on https (Caddy Server proxy to port 8080)

What component does this PR affect? 

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [x] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

Issues: not that I have known

close/fix #582 

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Docs update
- [x] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

**Comment by iUtsavB** [comment]
done testing, all okay, you can merge now

File: docs/graphql/manual/deployment/digitalocean/index.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+- Click on create droplet then on ``One-click apps``
+- Choose ``Docker`` from the ``One-click apps`` list
+- Choose your droplet size
+- Choose nearest datacenter region from your location
+- Select your ssh key add if you haven't added
+- Choose a hostname and then click on create button.
+- Now SSH into the droplet.
+- Run ``apt install update && apt upgrade -y`` to update system 
+
+Step 2: Get the Docker deployment and Caddy file
```
Comment (shahidhk) [inline docs/graphql/manual/deployment/digitalocean/index.rst]: This step can be renamed as `Get required setup files`

File: docs/graphql/manual/deployment/digitalocean/index.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+contains all installation manifests required to deploy Hasura on DigitalOcean.
+
+Get the docker deployment and Caddy file from there:
+
+.. code-block:: bash
+
+   $ wget https://raw.githubusercontent.com/hasura/graphql-engine/master/install-manifests/docker-compose-https/docker-compose.yaml
+   $ wget https://raw.githubusercontent.com/hasura/graphql-engine/master/install-manifests/docker-compose-https/Caddyfile
+
+Step 3: Set an access key and point domain to the public ip of the droplet.
```
Comment (shahidhk) [inline docs/graphql/manual/deployment/digitalocean/index.rst]: This step can be an optional one for setting up HTTPS. We can word it such that a user who doesn't need HTTPS can skip this.

Adding access key can be moved to a step towards the end under section `Securing the GraphQL Endpoint`

File: docs/graphql/manual/deployment/digitalocean/index.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      HASURA_GRAPHQL_DATABASE_URL: postgres://postgres:@postgres:5432/postgres
+    command:
+  ...
+
+Now replace ``<your-domain.com>`` in ``Caddyfile`` with your domain name 
+
+Example ``example.org``
+
+
+Step 4: Run the hasura docker container
```
Comment (shahidhk) [inline docs/graphql/manual/deployment/digitalocean/index.rst]: Title can be renamed as `Start GraphQL Engine`

We also need to add a section on using with existing Postgres.

File: docs/graphql/manual/deployment/digitalocean/index.rst (hunk 2 above)
Comment (iUtsavB) [reply docs/graphql/manual/deployment/digitalocean/index.rst]: then the user needs to download another .yaml file, isn't it?

Comment (shahidhk) [reply docs/graphql/manual/deployment/digitalocean/index.rst]: Yes, if the user doesn't need HTTPS, they can just download https://github.com/hasura/graphql-engine/blob/master/install-manifests/docker-compose/docker-compose.yaml

Maybe you can have 2 sub-sections: with/without HTTPS.

Comment (iUtsavB) [reply docs/graphql/manual/deployment/digitalocean/index.rst]: got it. 👍

**Comment by shahidhk** [comment]
LGTM. @rikinsk Needs review.

**Comment by rikinsk** [comment]
Moved the guide to guides section. 

@shahidhk LGTM

**Comment by shahidhk** [comment]
@rikinsk Since there is already a deploying section in docs, do you think people will miss the guides section for deployment stuff?

**Pull request merged**